examples_tests/13.BRDF_Explorer/BRDFExplorerApp.cpp

```diff
@@ -506,29 +506,35 @@ void BRDFExplorerApp::eventAOTextureBrowse(const ::CEGUI::EventArgs&)
         const auto image = ext::cegui::loadImage(p.second.c_str());
         loadTextureSlot(ETEXTURE_SLOT::TEXTURE_AO, image.buffer, image.w, image.h);
 
+        char buffer[1024];
+        sprintf(buffer, "%s (%ix%i)\nLeft-click to select a new texture.",
+            p.second.c_str(), image.w, image.h);
+
         box->setText(p.second);
         updateTooltip(
             "MaterialParamsWindow/AOWindow/ImageButton",
-            ext::cegui::ssprintf("%s (%ix%i)\nLeft-click to select a new texture.",
-                p.second.c_str(), image.w, image.h)
-                .c_str());
+            buffer);
     }
 }
 
 void BRDFExplorerApp::eventAOTextureBrowse_EditBox(const ::CEGUI::EventArgs&)
 {
     auto box = static_cast<CEGUI::Editbox*>(
         GUI->getRootWindow()->getChild("MaterialParamsWindow/AOWindow/Editbox"));
-
-    if (ext::cegui::Exists(box->getText().c_str())) {
+    
+    auto fs = device->getFileSystem();
```

Comment:
Where does `device` come from?

Reply:
Hmmmmmm... Now that I've thought about it, you'd have to have a `IrrlichtDevice*` available to use Irrlicht's filesystem-related utilities. And that's not nice considering `device` lives at `main()`. Suggestions?

Reply:
IMO helpers should be a standalone piece of function as-is, not having to depend on `IrrlichtDevice*` or anything else. But it's you guys' decision.

Reply:
BRDFExplorerApp can keep an IrrlichtDevice around

Reply:
Yeah, add new member to the app, you can refer to my latest commit https://github.com/buildaworldnet/IrrlichtBAW/pull/291/commits/d51bb8e8c3cf201fc90c4b17ed7ea27348d8c37f

Reply:
Remember the forward declaration (use it as much as possible).

Reply:
@Florastamie do you still have questions regarding this?

Reply:
It's all good